server/src/main/java/com/defold/extender/Extender.java

```diff
@@ -2094,7 +2094,9 @@ private File buildMainDexList(List<String> jars) throws ExtenderException {
         try {
             mainList.createNewFile();
             for (String classFile : mainClassNames) {
-                FileUtils.writeStringToFile(mainList, classFile + "\n", Charset.defaultCharset(), true);
+                // create main dex list in form of Proguards rules. Additional info https://github.com/defold/extender/issues/393
+                classFile = classFile.replace("/", ".").replace(".class", "");
+                FileUtils.writeStringToFile(mainList, String.format("-keep class %s { *; }\n", classFile), Charset.defaultCharset(), true);
```

Comment:
Create main dex list in form of Proguard rules.

Reply:
Does it lead to increasing the build size? Or is it merely a different syntax?

Reply:
Different syntax. Here is nothing happened regarding build shrinking.